kratos/input_output/stl_io.cpp

```diff
@@ -95,12 +89,7 @@ void StlIO::WriteEntityBlock(const TContainerType& rThisEntities)
 void StlIO::WriteGeometryBlock(const GeometriesMapType& rThisGeometries)
 {
     for (auto & r_geometry : rThisGeometries) {
-        // restrict to triangles only for now
-        const bool is_triangle = (
-            r_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D3 ||
-            r_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D6);
-
-        if (is_triangle) {
+        if (IsValidGeometry(r_geometry)) {
             WriteFacet(r_geometry);
```

Comment:
Maybe we can throw a warning if this happens. At least in debug mode.

Reply:
Warning added!